docs/proposal-006-build-improvements.md

```diff
@@ -0,0 +1,55 @@
+# Design Proposal #006: Build Improvements
+
+## Current State
+
+At the start of the project we've picked CircleCI as it appears to be a default choice at Weaveworks.
+
+The original solution involved:
+- running `docker build` to create a build image
+- use build image to run tests produce final image
+
+Publishing build and final images hasn't been automated, but some users requested it.
+
+When we switched [to modules](https://github.com/weaveworks/eksctl/pull/917), we have increased the
+compexity of how build image worked. It's hard for a new person to understand it, e.g. for a new
+contributor.
+
+We currently build images in CI and run tests as part of that, we should only run tests as a primary
+task.
```

Comment:
Do you mean that build and running tests should be two separate build steps? Rather than one combined one?

Reply:
Yes, especially as there is overhead associated with building the image and requirement of being able to run any `docker` commands, vs just running a Docker image natively.